tests/route/test_route_flap.py

```diff
@@ -98,6 +98,96 @@ def get_ptf_recv_ports(duthost, tbinfo):
     return recv_ports
 
 
+def get_neighbor_info(duthost, dev_port, tbinfo):
+    """
+    This function returns the neighbor type of
+    the chosen dev_port based on route info
+
+    Args:
+        duthost: DUT belong to the testbed.
+        dev_port: Chosen dev_port based on route info
+        tbinfo: A fixture to gather information about the testbed.
+    """
+    neighbor_type = ''
+    config_facts = duthost.config_facts(host=duthost.hostname, source="running")['ansible_facts']
+    neighs = config_facts['BGP_NEIGHBOR']
+    dev_neigh_mdata = config_facts['DEVICE_NEIGHBOR_METADATA'] if 'DEVICE_NEIGHBOR_METADATA' in config_facts else {}
+    mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
+    nbr_port_map = mg_facts['minigraph_port_name_to_alias_map'] \
+        if 'minigraph_port_name_to_alias_map' in mg_facts else {}
+    for neighbor in neighs:
+        local_ip = neighs[neighbor]['local_addr']
+        nbr_port = get_port_by_ip(config_facts, local_ip)
```

Comment:
nbr_port could be Ethernet<> or PortChannel<>.. right?
how are you handling portchannel case here?

Reply:
'_dev_port_' which is passed to this method '_get_neighbor_info_' either can be Portchannel<> or Ethernet<>.

And __get_port_by_ip__ method above returns the nbr_port based on the dut config_facts of '_PORTCHANNEL_INTERFACE_' and '_INTERFACE_'. So, if the nbr_port returned by _get_port_by_ip_ method is a Portchannel<> which can be directly same as '_dev_port_' passed through '_get_neighbor_info_'. Using that, we are fetching the neighbor's name and its type.

But if the nbr_port returned by 'get_port_by_ip' is Ethernet<>, it is actually **alias interface** name, not the actual interface name. So, we get the actual interface name using the _port_name_to_alias_map_ from minigraph facts for fetching the neighbor's name and its type.

Reply:
got it. thanks for the explanation..